### ISSUE TITLE
PR #1789: XLMRoBERTaNER MoreData diverges (loss 2.84 -> 9.40)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1855]
## Summary
`XLMRoBERTaNER.MoreData_ShouldNotDegrade` fails: the model diverges with more training (loss more than triples from short to long training). Reproduces in isolation (not a CI flake).

## Failing test (shard: ModelFamily - Generated Layers T-Z)
- `XLMRoBERTaNERTests.MoreData_ShouldNotDegrade`

## Error
```
2-iteration clone loss (9.396382) > 1-iteration loss (2.839479) - measured via GetLastLoss
(the model's own CE over logits). Optimizer may be diverging with more training.
```
The NER head's cross-entropy grows with more optimizer steps — a learning-rate / gradient-scale instability or a tape issue in the token-classification head. Compare against XLM-RoBERTa NER conventions (HuggingFace) and the loss/logit contract.

## Repro
```
dotnet test tests/AiDotNet.Tests/AiDotNetTests.csproj --framework net10.0 -c Release --no-build \
  --filter "FullyQualifiedName~XLMRoBERTaNERTests.MoreData_ShouldNotDegrade"
```

## Classification
Real pre-existing model bug ([51 s] in isolation). Fix training stability per paper/HF; do not weaken the test.

**Comment by ooples** [comment]
Verified fixed on PR #1789 — commit `7bbeea119`: `TransformerNERBase` now trains with the model's configured optimizer instead of the base default. Now linked with `Closes #1855`.
